### PR TITLE
Add system assigned managed identity to vm

### PR DESCRIPTION
**Pull request by zhengchang907**
-- Update UI to expose configuration options to users
-- Update mainTemplate.json to accept&pass parameters around
-- Update adminTemplate.json/adminTemplateForCustomSSL.json to add configuration for identity property

### CODE REVIEW

**Comment by zhengchang907** [comment]
@edburns Hi Ed, could you please review this PR? I've checked that it can be merged with the other [PR(Add Key Vault automation)](https://github.com/wls-eng/arm-oraclelinux-wls-admin/pull/77) without conflicts.

File: src/main/arm/createUiDefinition.json [hunk 1]
```diff
@@ -245,6 +245,27 @@
...
+                                {
+                                    "label": "No",
+                                    "value": false
+                                }
+                            ],
+                            "required": true
+                        },
+                        "defaultValue": "Yes",
+                        "visible": "[bool(basics('basicsOptional').basicsOptionalAcceptDefaults)]"
+                    },
```
Comment (edburns) [inline src/main/arm/createUiDefinition.json]: I am not sure why you did not use the same configuration I offered in my branch on `arm-oraclelinux-wls`:

```json
                    {
                        "name": "useSystemAssignedManagedIdentity",
                        "label": "Cause a system assigned managed identity to be created for the VM(s).",
                        "type": "Microsoft.Common.CheckBox",
                        "toolTip": "System assigned managed identities enable credential-free secure access to many Azure resources from this VM.",
                        "defaultValue": true
                    },            
...
            "useSystemAssignedManagedIdentity": "[basics('basicsOptional').useSystemAssignedManagedIdentity]",        
```

A checkbox is a better UI element than Yes/No for this use case.

**Review by edburns — CHANGES_REQUESTED** [review]
See comment on `createUiDefinition`.

**Comment by edburns** [comment]
Hello @zhengchang907 , I tried to resolve the merge conflicts, but was unable to cleanly do so. Can you please update the PR and resolve the conflicts so it will merge cleanly?

**Comment by zhengchang907** [comment]
Hi @edburns , I'll close this PR and re-create one with changes of System Assigned Managed Identity only.

I compared the src/main/arm/createUiDefinition.json line by line, the only difference is the system identity checkbox, the Github is acting weird.

**Pull request closed without merging**